### PR TITLE
TST: add a test on mixed offsets for read_csv

### PR DESCRIPTION
**Pull request by natmokval**
xref #54014

added a test on mixed offsets to increase coverage for `read_csv`

### CODE REVIEW

**Comment by natmokval** [comment]
> test looks good, but I think `pandas/tests/io/parser/test_parse_dates.py` might be a better location

Thanks, I moved the test as you suggested.

File: pandas/tests/io/parser/test_parse_dates.py [hunk 1]
```diff
@@ -2237,3 +2237,17 @@ def test_parse_dates_arrow_engine(all_parsers):
         }
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_from_csv_with_mixed_offsets():
```
Comment (MarcoGorelli) [inline pandas/tests/io/parser/test_parse_dates.py:2242]: does it work to use the
```
parser = all_parsers
```
and then
```
parser.read_csv
```
pattern from the other tests in this file?

Comment (natmokval) [reply pandas/tests/io/parser/test_parse_dates.py]: Yes, it works very well, except `pyarrow`. I added `@xfail_pyarrow`.

**Review by MarcoGorelli — APPROVED** [review]
looks good to me on green

**Comment by natmokval** [comment]
Thank you for reviewing this PR.

**Pull request merged**